examples/hosted-wallet-ts/src/pfi/api/exchanges.ts

```diff
@@ -0,0 +1,363 @@
+    res.send(responseBody.toJSONString());
+  });
+
+  /**
+   * POST /exchanges
+   * Handles creation of a new exchange (RFQ)
+   */
+  router.post('/', async (req, res) => {
+    console.log('POST /exchanges');
+
+    try {
+      const createExchangeRequestBody = CreateExchangeRequestBody.fromJSONString(
+        JSON.stringify(req.body)
+      );
+
+      const rfq: Rfq = createExchangeRequestBody.message;
+      await rfq.verify();
+
+      const offering = offeringsRepository.getOffering(rfq.data.offeringId);
+      if (!offering) {
+        throw new Error('Offering not found');
+      }
+
+      // TODO: Uncomment when we have valid pex input
+      // await rfq.verifyOfferingRequirements(offering);
+
+      if (createExchangeRequestBody.replyTo) {
+        exchangesToReplyTo.set(rfq.metadata.exchangeId, createExchangeRequestBody.replyTo);
+      }
+
+      exchangeIdToExchange.set(rfq.metadata.exchangeId, { rfq });
+
+      // Asynchronously send a quote in response to the RFQ
+      setImmediate(async () => {
+        try {
+          await replyWithQuote(rfq.metadata.from, rfq.metadata.exchangeId);
+        } catch (error) {
+          console.error('Error in replyWithQuote:', error);
+        }
+      });
+
+      res.status(202).send();
+    } catch (error) {
+      console.error('Error in POST /exchanges:', error);
+      res.status(400).json({
+        error: error.message || 'RFQ does not satisfy an available offering',
+      });
+    }
+  });
+
+  /**
+   * PUT /exchanges/:id
+   * Updates an existing exchange with a new message (Order or Cancel)
+   */
+  router.put('/:id', async (req, res) => {
+    const exchangeId = req.params.id;
+    console.log(`PUT /exchanges/${exchangeId}`);
+
+    try {
+      const updateExchangeRequestBody = UpdateExchangeRequestBody.fromJSONString(
+        JSON.stringify(req.body)
+      );
+
+      const message = updateExchangeRequestBody.message;
+
+      if (message instanceof Order) {
+        const orderMessage = message as Order;
+        await orderMessage.verify();
+
+        // Process the order asynchronously
+        await processOrder(orderMessage);
+      } else if (message instanceof Cancel) {
+        const cancelMessage = message as Cancel;
+        await cancelMessage.verify();
+
+        // Process the cancellation
+        await replyWithClose(cancelMessage.metadata.from, cancelMessage.metadata.exchangeId, false);
+      }
+
+      res.status(202).send();
+    } catch (error) {
+      console.error('Error in PUT /exchanges/:id', error);
+      res.status(400).json({ error: error.message || 'Invalid request' });
+    }
+  });
+
+  /**
+   * Processes an Order message by sending a series of status updates
+   * @param orderMessage - The Order message to process
+   */
+  async function processOrder(orderMessage: Order) {
+    try {
+      await replyWithOrderInstructions(orderMessage.metadata.from, orderMessage.metadata.exchangeId);
+
+      const statuses = [
```

Comment:
this makes me wonder, should we add well-defined order status values inside the sdk?

Reply:
yes definitely, I was wondering if I was using this wrong